### PR TITLE
[TOOLS-517] Taskboard and dashboard improvements

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: src/main/resources/static/scripts/dc-utils.js [hunk 1]
```diff
@@ -10,34 +10,96 @@ function DcUtils() {
...
+            numOfTicks = 2;
+
+        var cols = numOfTicks - 1;
+
+        var oneDayInMilli = 86400000;
+        var diffInDays = (endDate - startDate)/oneDayInMilli;
+        var betweenTicks = diffInDays / cols;
+
+        var ticks = [];
+        for (i = 0; i < cols; i++) {
```
Comment (objective-solutions-user) [inline src/main/resources/static/scripts/dc-utils.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "i" to make it explicit. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AS2703)

File: src/main/resources/static/scripts/tb-pie-chart.js [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+                    highlightLabel(i, false);
+                });
+
+            _chart._applyLabelText(labelsText);
+            positionLabels(labelsGroups, arc);
+            if (_externalLabelRadius && _drawPaths) {
+                updateLabelPaths(pieData, arc);
+            }
+
+            labelsText.each(function(d, i) {
```
Comment (objective-solutions-user) [inline src/main/resources/static/scripts/tb-pie-chart.js]: MAJOR Remove the unused function parameter "i". [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: src/main/resources/static/scripts/tb-pie-chart.js [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+        } else {
+            _chart.selectAll('g.' + _sliceCssClass).each(function () {
+                _chart.resetHighlight(this);
+            });
+        }
+    }
+
+    function highlightLabels () {
+        if (_chart.hasFilter()) {
+            _chart.selectAll('g.' + _tbLabelCssClass).each(function (d, i) {
```
Comment (objective-solutions-user) [inline src/main/resources/static/scripts/tb-pie-chart.js]: MAJOR Remove the unused function parameter "i". [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: src/main/resources/static/scripts/tb-pie-chart.js [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+ */
+dc.tbPieChart = function (parent, chartGroup) {
+    var _tbLabelCssClass = 'tb-label',
+        _tbSelectedClass = 'selected',
+        _tbDeselectedClass = 'deselected';
+
+    var DEFAULT_MIN_ANGLE_FOR_LABEL = 0.5;
+
+    var _sliceCssClass = 'pie-slice';
+    var _labelCssClass = 'pie-label';
```
Comment (objective-solutions-user) [inline src/main/resources/static/scripts/tb-pie-chart.js]: MAJOR Remove this useless assignment to local variable "_labelCssClass" [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AS1854)
MINOR Remove the declaration of the unused '_labelCssClass' variable. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AUnusedVariable)

File: src/main/resources/static/scripts/tb-pie-chart.js [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+    };
+
+    _chart.legendToggle = function (d) {
+        _chart.onClick({key: d.name, others: d.others});
+    };
+
+    function highlightSliceFromLegendable (legendable, highlighted) {
+        _chart.selectAll('g.pie-slice').each(function (d, i) {
+            if (legendable.name === d.data.key) {
+                //d3.select(this).classed('highlight', highlighted);
```
Comment (objective-solutions-user) [inline src/main/resources/static/scripts/tb-pie-chart.js]: MAJOR Remove this commented out code. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 6 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/resources/static/elements/taskboard/kpis/widget-scope-by-type.html [hunk 6]
```diff
@@ -130,23 +128,30 @@
...
                 },
 
                 _getChartColors: function() {
-                    var baselineBacklog = '#909',
-                        baselineDone = '#2ca02c',
-                        intangible = '#dc3912',
-                        newScope = '#1f77b4',
-                        rework = '#ff7f0e';
-                    return dcUtils.getChartColorsFromArray([baselineBacklog, baselineDone, intangible, newScope, rework]);
+                    var baselineBacklog = dcUtils.colors.BASELINE_BACKLOG,
```
Comment (taksan) [inline src/main/resources/static/elements/taskboard/kpis/widget-scope-by-type.html]: Não entendi o porque das variáveis ao invés de colocar direto no array

**Pull request merged**